### ISSUE TITLE
Paper size change behavior at v1.1.0

### ISSUE THREAD

**Issue by jnbarchan** [clientIO/joint#574]
There is some change in behaviour of Paper size at v1.1.0 which is breaking my working code from v1.0.3.  Tested under IE 11, Firefox & Chrome.

The outline of my code is:

    <div id="divPaper" style="width: 640px; height: 590px; overflow-x: hidden; overflow-y: scroll;">
    </div>

    var $divPaper = $('#divPaper');
    var paper = new joint.dia.Paper({
      el: $divPaper,
      width: $divPaper[0]['clientWidth'] - 5,	// excludes any scrollbar, -5 for some rounding error I found
      height: $divPaper[0]['clientHeight'] - 5,	// excludes any scrollbar, -5 for some rounding error I found
      model: graph
    });

I have to have a scrollbar to the right of the paper.  I believe my use of `clientWidth/Height` is conceptually correct --- that excludes scrollbars & margins, and I want the paper to fill that area.

Immediately prior to the `new Paper` statement, `clientWidth=623, clientHeight=590` in both versions.  This seems correct, as the vertical scroller is 17px wide.

In v1.0.3 both measurements _remain unchanged_ after the Paper is created.  That is what I expect and seems correct.

In v1.1.0 `clientWidth=601, clientHeight=585` after the Paper is created.  This is too small, moving my scrollbar over the content.

What is the change in v1.1.0 which is causing this change in behaviour?  It seems to be altering my enclosing container DIV to change size!  Which of v1.0.3 or v1.1.0 is "correct" or "intended"?  If v1.1.0 is correct, how must I change my code to compensate?

**Comment by jnbarchan** [comment]
I have done some further playing, trying to find a workaround but not succeeding.  Something has been changed at v1.1.0, such that the enclosing container DIV is being resized (by altering its `style="width: ...; height: ...;"`) as the JointJS Paper changes size, even at runtime.  Making it impossible for the container to have scrollbar(s), which was all working fine at v1.0.3?

**Comment by jnbarchan** [comment]
I'm impatient, so I had a look at differences between v1.0.3 & v1.1.0.  Would I be right in saying that the cause of this change of behaviour is that in `Paper.setDimensions()`:

In v1.0.3:

    line 9595: V(this.svg).attr({ width: width, height: height });

In v1.1.0:

    line 10304:         this.$el.css({
            width: Math.round(width),
            height: Math.round(height)
        });

So you used to set dimensions on the `<svg>` element but now you set them on the container element (my `<div>`)?  But you haven't told us of that change?  And because it sets the container's width/height to exactly the Paper's width/height there can no longer be any room for any container "furniture"?

**Comment by kumilingus** [comment]
Could you wrap the paper with a container element instead?
```javascript
var paperContainer = document.getElementByID('paperContainer');
var paper = new joint.dia.Paper({
  width: divPaperContainer.clientWidth - 5,
  height: divPaperContainer.clientHeight - 5,
  model: graph
});
paperContainer.appendChild(paper.el);
```

**Comment by jnbarchan** [comment]
Your post beat me to it!!

I just came here to say I have amended my HTML (only, no change to my code) to:

    <div id="divPaperWrapper" style="width: 640px; height: 590px; overflow-x: hidden; overflow-y: scroll;">
      <div id="divPaper" style="width: 100%; height: 100%;">
      </div>
    </div>

So you are changing `divPaper`'s width/height, and I have moved the size & scrolling out to `divPaperWrapper`.  Seems to work OK.  You would want me to change over to this, right?

You _might_ want to document somewhere that you are now taking control of the outside world's container's width/height, because this was a change in behaviour.  May I ask why you did it, what you wanted to achieve?  (Was it to do with introducing background colouring?)
